### PR TITLE
Template generator

### PR DESCRIPTION
**Pull request by elfiesmelfie**
Updated the template generator to use ansible and render the defaults and the plugin.conf.j2 files

Use with::

ansible-playbook tools/template_generator.yml 

Still can't do complex configs, but most collectd configs have simple parameters anyway (string, number, bool, list), and only a minority have complex configs that require parsing dictionaries.

### CODE REVIEW

File: tools/README [hunk 1]
```diff
@@ -1,9 +1,16 @@
-Template generate
------------------
+Template generater
```
Comment (mrunge) [inline tools/README]: generator

File: .travis.yml [hunk 2]
```diff
@@ -5,11 +5,17 @@ git:
 addons:
   apt:
     update: true
+services:
+  - docker
 before_install:
-  - sudo apt-get -y install collectd
+  - sudo docker pull opnfv/barometer
```
Comment (mrunge) [inline .travis.yml]: can we please pull from tripleomaster/centos-binary-collectd

**Review by leifmadsen — APPROVED** [review]

**Comment by leifmadsen** [comment]
@elfiesmelfie this is still open (didn't notice it until now :)) do you still want this to merge down? We'll just need to make sure we adjust the .travis.yml and rebase this branch against master once the roles-import branch merges (assuming it does).

**Comment by leifmadsen** [comment]
Created PR for the collectd_config changes here:  https://github.com/infrawatch/collectd-config-ansible-role/pull/4

**Comment by leifmadsen** [comment]
Updated this branch to match master work. Changes to the collectd_config role are in the infrawatch/collectd-config-ansible-role repo now.

**Comment by elfiesmelfie** [comment]
> Updated this branch to match master work. Changes to the collectd_config role are in the infrawatch/collectd-config-ansible-role repo now.

Ack, I can see that in [1].
[1] https://github.com/infrawatch/collectd-config-ansible-role/pull/4/files

**Pull request merged**